### PR TITLE
[WFLY-14968] Collapse if-statements (EJB)

### PR DESCRIPTION
**Pull request by boris-unckel**
Fixes https://issues.redhat.com/browse/WFLY-14968
PRs splitted in components, here first one EJB.

### CODE REVIEW

**Comment by bstansberry** [comment]
@chengfang please review

File: ejb3/src/main/java/org/jboss/as/ejb3/timerservice/TimerServiceImpl.java [hunk 1]
```diff
@@ -408,20 +408,18 @@ public TimerImpl loadAutoTimer(ScheduleExpression schedule,
...
             for (final TimerImpl timer : this.timers.values()) {
                 // Less disruptive way to get WFLY-8457 fixed.
-                if (timer.isActive() || (!timer.isActive() && timer.getState() == TimerState.ACTIVE)) {
-                    if (timer.getPrimaryKey() == null || timer.getPrimaryKey().equals(pk)) {
-                        activeTimers.add(timer);
-                    }
+                if ((timer.isActive() || !timer.isActive() && timer.getState() == TimerState.ACTIVE)
+                        && (timer.getPrimaryKey() == null || timer.getPrimaryKey().equals(pk))) {
+                    activeTimers.add(timer);
                 }
```
Comment (chengfang) [inline ejb3/src/main/java/org/jboss/as/ejb3/timerservice/TimerServiceImpl.java:414]: The first segment of the above if condition:
`(timer.isActive() || !timer.isActive() && timer.getState() == TimerState.ACTIVE)` can be simplified to:
`(timer.isActive() || timer.getState() == TimerState.ACTIVE)`

Comment (boris-unckel) [reply ejb3/src/main/java/org/jboss/as/ejb3/timerservice/TimerServiceImpl.java:414]: @chengfang Hello Cheng, I have fixed it. May you review and approve, please?

**Review by chengfang — APPROVED** [review]

**Comment by boris-unckel** [comment]
Hello @bstansberry , this PR is also ready. Please, could you check and merge?

**Pull request merged**